### PR TITLE
Feature/first sight

### PR DESCRIPTION
**Pull request by digisomni**
This PR is based on PR #10.

### CODE REVIEW

**Comment by digisomni** [comment]
This should be merged into master before continuing work on TypeScript support, that way everything can be migrated all at once.

File: src/components/MainScene.vue [hunk 1]
```diff
@@ -61,17 +59,96 @@ export default {
...
+
+            this.$store.state.Entities.addEntity(this.scene, {
+                'name': 'box',
+                'type': 'Shape',
+                'shape': 'box',
+                'position': { 'x': -5, 'y': 0, 'z': 0 },
+                'rotation': { 'x': -0.2, 'y': -0.4, 'z': 0 },
+                'dimensions': { 'x': 3, 'y': 3, 'z': 3 },
+                'color': { 'r': 1, 'g': 0, 'b': 0 }
+            }, {});
```
Comment (ctrlaltdavid) [inline src/components/MainScene.vue:73]: Shouldn't really use strings for property IDs in script. This and similar could be written...
```suggestion
            this.$store.state.Entities.addEntity(this.scene, {
                name: 'box',
                type: 'Shape',
                shape: 'box',
                position: { x: -5, y: 0, z: 0 },
                rotation: { x: -0.2, y: -0.4, z: 0 },
                dimensions: { x: 3, y: 3, z: 3 },
                color: { r: 1, g: 0, b: 0 }
            }, {});
```

File: src/modules/entities/entities.js [hunk 2]
```diff
@@ -8,31 +8,114 @@
...
+                BABYLON.SceneLoader.ImportMeshAsync(name, urlWithoutFilename, filename, scene).then((result) => {
+                    resolve(result.meshes[0]);
+                });
+            });
+        };
     }
 
-    return {
-        add
+    async addEntity (scene, properties, entityHostType) {
```
Comment (ctrlaltdavid) [inline src/modules/entities/entities.js:34]:
```suggestion
    async addEntity(scene, properties, entityHostType) {
```

File: src/modules/entities/entities.js [hunk 3]
```diff
@@ -8,31 +8,114 @@
...
+        if (entityToDelete) {
+            entityToDelete.dispose();
+        } else {
+            Debug.error('ENTITIES', 'Failed to delete entity by ID: ' + id);
+        }
+    };
+
+    // Scripting would look like:
+    // Entities.deleteByName
+    async deleteEntityByName (scene, name) {
```
Comment (ctrlaltdavid) [inline src/modules/entities/entities.js:112]:
```suggestion
    async deleteEntityByName(scene, name) {
```

File: src/modules/entities/entities.js [hunk 4]
```diff
@@ -8,31 +8,114 @@
...
+        } else {
+            Debug.error('ENTITIES', 'Failed to create entity.');
+        }
     };
-}());
 
-export default Entities;
+    // Scripting would look like:
+    // Entities.deleteById
+    async deleteEntityById (scene, id) {
```
Comment (ctrlaltdavid) [inline src/modules/entities/entities.js:100]:
```suggestion
    async deleteEntityById(scene, id) {
```

File: src/modules/debugging/debug.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+//  Copyright 2021 Vircadia contributors.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+*/
+
+import Log from './log.js';
+
+const Debug = (function () {
+    function error (type, string) {
```
Comment (ctrlaltdavid) [inline src/modules/debugging/debug.js:14]:
```suggestion
    function error(type, string) {
```

File: src/modules/metaverse/metaverse.js [hunk 6]
```diff
@@ -175,6 +174,28 @@ export class Metaverse {
                 });
         });
     };
-    
+
     // #endregion Login & Registration
+
+    // #region Helpers
+
+    getError (error, stringify) {
```
Comment (ctrlaltdavid) [inline src/modules/metaverse/metaverse.js:182]:
```suggestion
    getError(error, stringify) {
```

File: src/modules/metaverse/metaverse.js [hunk 7]
```diff
@@ -175,6 +174,28 @@ export class Metaverse {
...
+
+        if (error.response && error.response.data) {
+            toReturn = error.response.data;
+        } else if (error.response) {
+            toReturn = error.response;
+        } else {
+            toReturn = error;
+        }
+
+        if (stringify === true) {
```
Comment (ctrlaltdavid) [inline src/modules/metaverse/metaverse.js:193]:
```suggestion
        if (stringify) {
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

File: src/components/MainScene.vue [hunk 8]
```diff
@@ -61,17 +59,96 @@ export default {
...
+
+            this.$store.state.Entities.addEntity(this.scene, {
+                'name': 'sphere',
+                'type': 'Shape',
+                'shape': 'sphere',
+                'position': { 'x': -3, 'y': 0, 'z': 0 },
+                'rotation': { 'x': 0, 'y': -0.5, 'z': 0 },
+                'dimensions': { 'x': 3, 'y': 3, 'z': 3 },
+                'color': { 'r': 0, 'g': 0.58, 'b': 0.86 }
+            }, {});
```
Comment (ctrlaltdavid) [inline src/components/MainScene.vue:83]: Similar changes to be made here and below.

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Review by ctrlaltdavid — APPROVED** [review]

**Pull request merged**